server/CVCMIServer.cpp

```diff
@@ -279,33 +279,54 @@ void CVCMIServer::prepareToRestart()
 
 bool CVCMIServer::prepareToStartGame()
 {
+	Load::ProgressAccumulator progressTracking;
+	Load::Progress current(1);
+	progressTracking.include(current);
+	
+	auto progressTrackingThread = boost::thread([this, &progressTracking]()
+	{
+		while(!progressTracking.finished())
+		{
+			std::unique_ptr<LobbyLoadProgress> loadProgress(new LobbyLoadProgress);
+			loadProgress->progress = progressTracking.get();
+			addToAnnounceQueue(std::move(loadProgress));
+			boost::this_thread::sleep(boost::posix_time::milliseconds(50));
+		}
+	});
```

Comment:
Would be great to have an option to install callback that notifies whenever loading progress changes instead of yet another thread